### DISCUSSION TITLE
Some confusion about the performance issue

### DISCUSSION THREAD

**Discussion by MadFlyFish** [kagenash1/godot-behavior-tree#18] [General]
When the number of BTnodes and Agents increases, would performance become a critical problem?
I would appreciate it if you have time to see this post.

**Comment by kagenash1** [comment]
Honestly I haven't found issues with it, not more than what happens normally in Godot (so performance hit would be significant in the order of 100,000 nodes). Obviously this is assuming you are being very careful with your code and avoiding unnecessary operations. 
This is a minimal implementation which mostly uses Godot API things so it's as fast as what you normally do with any other node.
Personally I've tried having enemies with 5/6 concurrent behaviors involving pathfinding, detecting targets and covers, shooting, taking cover, moving in constantly changing patterns and putting hundreds of them into the battlefield all at once, all while having all the other mechanics of the game (a top down shooter). And this was back when the plugin was waaaay less optimised and even so I didn't lag a single FPS. 
So ultimately don't worry about it as long as you optimize as best as you can (meaning use the engine API everytime you can and avoid complex logic with GDScript, but this is true always, even outside of this plugin) and take advantage of the "tree suspension" with yields to avoid unnecessary ticks. Cheers :)

**Reply by MadFlyFish** [reply]
Thanks. It's a good guide to me.
It seems to fit the needs of my project quite well. :)
